frontend/src/game/jp/controller/JapaneseRound.ts

```diff
@@ -257,10 +254,17 @@ const isJapaneseGameEnd = (
     if (!exceedsHanten) {
         return false;
     }
-    if (newRound.roundWind === Wind.EAST || newRound.roundWind === Wind.SOUTH) {
-        return false;
+    // At least one person is more than 30k
+    if (newRound.roundWind === Wind.WEST) {
+        return true; // dealership gone; someone's more than 30k
+    }
+    const lastRound = concludedRounds[concludedRounds.length - 1];
+    if (lastRound.roundWind !== Wind.SOUTH || lastRound.roundNumber !== NUM_PLAYERS) {
+        return false; // not even S4 yet
     }
-    return true; // west, and one person's score exceeds 30k
+
+    totalScore[NUM_PLAYERS - 1] -= 1; // for tiebreaking purposes
```

Comment:
There's no reason to do the calculations this way, it's too hard to understand compared to a single loop